### DISCUSSION TITLE
Bugs related to Puzzles.

### DISCUSSION THREAD

**Discussion by shuvashish76** [gbtami/pychess-variants#1143] [General]
Announcement :https://www.pychess.org/news/Summer_Update

https://www.pychess.org/puzzle/1C3KZ
Solution given: Ef2+, Kg1 Ee2+, Kf1 Ee3+, Kf2 Ee2+
Which is wrong because there is a simple perpetual exist. Fairy-Stockfish gives 0.0 on my android browser.

Also noticed another bug
 - Copy-paste your solved puzzle link or refresh browser & solve again it counts that as a new puzzle which gives unfair ratings.

**Comment by gbtami** [comment]
This is a so called "defensive" puzzle. You lose the game if you are not giving the perpetual checks. So the solution is the best in this position.
Regarding ratings, this is similar on lichess. Solving puzzles  is not  a competition. Abusing puzzle rating is simply meaningless.

**Reply by shuvashish76** [reply]
> This is a so called "defensive" puzzle.

I think there are no "perpetual" puzzles on lichess.
Though I agree including "perpetuals" in Pychess makes sense as it's not always easy to find perpetuals in various variants.

> This is similar on lichess. Solving puzzles is not a competition. Abusing puzzle rating is simply meaningless.

Agreed. CMIIW but lichess doesn't add ratings for repeated puzzles either.

**Reply by gbtami** [reply]
I checked it now, and you are correct. Seems they notice that you solved it already and don't rate it again.
Possible this deserves to file a new issue :)
